src/components/Disputes/DisputeAutoReveal.js

```diff
@@ -0,0 +1,119 @@
+import React, { useCallback, useEffect, useRef, useState } from 'react'
+import { Button, GU, Info } from '@aragon/ui'
+import { useWallet } from '../../providers/Wallet'
+
+import { getAutoRevealRequest } from '../../services/autoReveal'
+import {
+  getAutoRevealPreference,
+  getCodeFromLocalStorage,
+  getOutcomeFromCommitment,
+} from '../../utils/crvoting-utils'
+
+function DisputeAutoReveal({ commitment, disputeId, onAutoReveal, roundId }) {
+  const { account } = useWallet()
+
+  const autoRevealRequested = useAutoRevealPolling(account, disputeId, roundId)
```

Comment:
Perhaps we should add a loading indicator and return null, to avoid a flash with the API.